File: strategy/marvin/test/observer/defense.lua
local DefenseTest = {}

local debug = require "../base/debug"
local vis = require "../base/vis"
local World = require "../base/world"
local Defense = require "util/defense"

function DefenseTest.testDangerousness()
	local ratings = Defense.rateOpponentDangerousness()
	for robot, rating in pairs(ratings) do
		vis.addCircle("test: Dangerousness", robot.pos, 0.2, vis.fromTemperature(rating), true)
	end
end

return DefenseTest

File: strategy/typescript/glados/test/observer/defense.ts
import * as vis from "base/vis";

import * as Defense from "glados/util/defense";

export function testDangerousness() {
	const ratings = Defense.rateOpponentDangerousness();
	for (const [robot, rating] of ratings.entries()) {
		vis.addCircle("test: Dangerousness", robot.pos, 0.2, vis.fromTemperature(rating), true);
	}
}

